sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerClient.java

```diff
@@ -259,9 +348,18 @@ public boolean exists() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Boolean> existsWithResponse(Duration timeout, Context context) {
-        Mono<Response<Boolean>> response = client.existsWithResponse(context);
-
-        return blockWithOptionalTimeout(response, timeout);
+        try {
+            Response<BlobContainerProperties> response = getPropertiesWithResponse(null, timeout, context);
+            return new SimpleResponse<>(response, true);
+        } catch (RuntimeException e) {
+            if (ModelHelper.checkContainerDoesNotExistStatusCode(e) && e instanceof HttpResponseException) {
```

Comment:
Looking at the [async version of this code](https://github.com/Azure/azure-sdk-for-java/blob/13cf877463b83121f43a4ee0e4b8727253594b62/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java#L365), it looks like it only checks for 404's? Is there a reason why we are not following the same pattern of just looking for 404's and instead looking at error codes as well?